EIPS/eip-1895.md

```diff
@@ -0,0 +1,158 @@
+---
+eip: 1895
+title: Support for an Elliptic Curve Cycles
+author: Alexandre Belling <alexandrebelling8@gmail.com>
+discussions-to: https://ethresear.ch/t/reducing-the-verification-cost-of-a-snark-through-hierarchical-aggregation/5128/7
+status: Draft
+type: Standards Track
+category: Core
+created: 2018-31-03
+---
+
+# Support for an Elliptic Curve Cycles
+
+## Simple Summary
+
+The EVM currently supports elliptic curves operations for curve *alt-bn128* thanks to precompiles `ecadd` and `ecmul` and `ecpairing`. The classes MNT4 and 6 contains cycles of curves and this enable doing operation on one curve inside a SNARK on the other (end reversely). This EIP suggests adding support for those curves.
```

Comment:
I'm not sure how to read this sentence:

_"The classes MNT4 and 6 contains cycles of curves and this enable doing operation on one curve inside a SNARK on the other (end reversely)."_

Could you perhaps clarify it?

Reply:
@AlexandreBelling Could you still clarify the sentence?

Reply:
Does it make more sense if I rephrase like that ?

"An elliptic curve cycle is a set of elliptic curves with the property that the inner-field of one curve is the outer-field of the other one. Cycles make it possible to do elliptic curve operations for one of the curves inside zkSNARK circuit that is proven and verified using the other curve of the cycle."

Reply:
That's an explanation, but the sentence structure is still unclear to me. It looks like the grammar is not correct. So you could leave the sentence in there, just fix the grammar. Perhaps ask an English native speaker to correct it.

Reply:
Hum ok, I have asked someone to help me fix it. I now believe it means what it's supposed to mean :)

Reply:
This might be "Two elliptic curves were generated such that group order of one curve is coordinates field characteristic of the other curve, and reverse."

We have a good arXiv paper referenced describing this property already, making it unreasonable explaining cycle here on github. I would also refer to my favorite IACR 2014/595 preprint.